### PR TITLE
Addition changes to enable Verify.Cli app

### PR DESCRIPTION
**Pull request by flcdrg**
- Add simple constructor to InnerVerifier for just comparing files
- Ensure replacements field is valid even if we haven't called UseAssembly

Assists with #716

### CODE REVIEW

**Comment by flcdrg** [comment]
Coming back to this, as https://github.com/flcdrg/Verify.Cli/ is just wanting to compare a file, it doesn't have the extra type stuff that the existing constructor requires (and trying to fake it gets messy).

This change, or something like it should enable file verification by being able to reference to the Verify nupkg

**Comment by flcdrg** [comment]
Hmm.. that change to ApplyScrubbers is breaking tests. I've got something wrong there.

**Comment by flcdrg** [comment]
I think this is ready to review. Mac build had some kind of hiccup and needs to be re-queued.

File: src/Verify/Serialization/Scrubbers/ApplyScrubbers.cs [hunk 1]
```diff
@@ -55,6 +55,8 @@ public static void UseAssembly(string? solutionDir, string projectDir)
         }
 
         AddProjectAndSolutionReplacements(solutionDir, projectDir, replacements);
+        //ApplyScrubbers.replacements.Clear();
```
Comment (SimonCropp) [inline src/Verify/Serialization/Scrubbers/ApplyScrubbers.cs:58]: whats with the commented code?

File: src/Verify/Verifier/InnerVerifier.cs [hunk 2]
```diff
@@ -70,6 +70,38 @@ public static void ThrowIfVerifyHasBeenRun()
         }
     }
 
+    public InnerVerifier(string sourceFile, VerifySettings settings)
```
Comment (SimonCropp) [inline src/Verify/Verifier/InnerVerifier.cs:73]: can u add some tests and docs for this

File: src/Verify/Serialization/Scrubbers/ApplyScrubbers.cs (hunk 1 above)
Comment (flcdrg) [reply src/Verify/Serialization/Scrubbers/ApplyScrubbers.cs:58]: Whoops!

File: src/Verify/Verifier/InnerVerifier.cs (hunk 2 above)
Comment (flcdrg) [reply src/Verify/Verifier/InnerVerifier.cs:73]: Will add them shortly

**Comment by flcdrg** [comment]
Test is failing.. trying to figure out why. I can't help but notice that there's two dots in the deleted output - that seems suspicious.

```
VerifyException: Directory: C:\dev\git\Verify\src\Verify.Tests

VerifyException
Directory: C:\dev\git\Verify\src\Verify.Tests
Delete:
  - sample.verified..txt
Equal:
  - Received: sample.txt
    Verified: sample.verified.txt
```

**Comment by SimonCropp** [comment]
https://github.com/VerifyTests/Verify/pull/888/files#r1205478076

File: src/Verify/Verifier/InnerVerifier.cs [hunk 3]
```diff
@@ -70,6 +70,42 @@ public static void ThrowIfVerifyHasBeenRun()
...
+            settings.namedDateTimes,
+            settings.namedGuids,
+            settings.namedDateTimeOffsets
+        );
+
+        IoHelpers.CreateDirectory(directory);
+
+        ValidatePrefix(settings, directory);
+
+        verifiedFiles = new List<string> { Path.Combine(directory, $"{Path.GetFileNameWithoutExtension(sourceFile)}.verified.{Path.GetExtension(sourceFile)}") };
```
[the remark this answers is not in the document]
Comment (flcdrg) [reply src/Verify/Verifier/InnerVerifier.cs]: Thanks, that did the trick!

**Pull request merged**